frontend/src/reducers/tables.js

```diff
@@ -137,11 +138,13 @@ const reducer = produce((draftState, action) => {
         }
       }
 
-      draftState[id] = {
-        ...prevTableStruct,
-        ...data,
-        ...updatedSelected,
-      };
+      draftState[id] = pending
+        ? { ...draftState[id], pending }
+        : {
+            ...prevTableStruct,
+            ...data,
+            ...updatedSelected,
+          };
```

Comment:
each time i stumble on this code.... i don't like it.

instead of doing this unreadable hack, why not making it obvious that in case of pending=true you don't give a s**t on data.rows....

so isn't better to change line 122 to reflect this, i.e.
```
if (!pending && data.rows && data.rows.length) {
```

so the change from here (line 141-147) won't be needed.

also in this way you would cover the case when, for some reason, draftState[id] is null/undefined.